frontend/coprs_frontend/coprs/logic/builds_logic.py

```diff
@@ -1132,15 +1132,19 @@ def finish(chroot, status):
 
                     exclusivearch = upd_dict["results"].get("exclusivearch")
                     if exclusivearch and arch not in exclusivearch:
-                        chroot.status_reason = \
-                            "This chroot was skipped because of ExclusiveArch"
-                        finish(chroot, StatusEnum("skipped"))
+                        # Skip only if all macros in ExclusiveArch evaluated
+                        if not [x for x in exclusivearch if x.startswith("%")]:
+                            chroot.status_reason = \
+                                "This chroot was skipped because of ExclusiveArch"
+                            finish(chroot, StatusEnum("skipped"))
 
                     excludearch = upd_dict["results"].get("excludearch")
                     if arch in excludearch:
-                        chroot.status_reason = \
-                            "This chroot was skipped because of ExcludeArch"
-                        finish(chroot, StatusEnum("skipped"))
+                        # Skip only if all macros in ExcludeArch evaluated
+                        if not [x for x in excludearch if x.startswith("%")]:
```

Comment:
```
In [1]: if [False, False]:
   ...:     print("true")
   ...: 
true

Reply:
But this is a different case

```python
>>> [x for x in ["foo", "bar", "baz"] if x.startswith("%")]
[]

>>> [x for x in ["foo", "%bar", "baz"] if x.startswith("%")]
['%bar']
```

The first one is `False`, the second one is `True`.

Reply:
ah, this is a check for emptiness (not obvious), then good!